src/components/profile/orderHistoryPage/Page.js

```diff
@@ -0,0 +1,221 @@
+import { useModal } from "@/components/_modal";
+import { orderStatus } from "@/lib/api/order.fragments";
+import { useEffect, useState } from "react";
+const itemsPerPage = 4;
+
+/**
+ * Shows the orders made by the user from bibliotekdk.
+ *
+ * @returns {component}
+ *
+ */
+
+export default function OrderHistoryPage() {
+  const { isAuthenticated } = useUser();
+  const breakpoint = useBreakpoint();
+  const modal = useModal();
+
+  const isMobile = breakpoint === "xs" || breakpoint === "sm";
+  const [orderDataPages, setOrderDataPages] = useState([]);
+  const [bibliotekDkOrderIds, setBibliotekDkOrderIds] = useState();
+  const [currentPageIndex, setCurrentPageIndex] = useState(0);
+  const [currentPageIds, setCurrentPageIds] = useState([]);
+
+  const { data: orderData, isLoading: orderDataIsLoading } = useData(
+    currentPageIds?.length > 0 && orderStatus({ orderIds: currentPageIds })
+  );
+  const { data: userData, isLoading } = useData(
+    isAuthenticated && orderHistory()
+  );
+
+  const fetchedOrders = orderData?.orderStatus;
+
+  const totalPages = Math.ceil(bibliotekDkOrderIds?.length / itemsPerPage);
+  useEffect(() => {
+    if (userData) {
+      const bibDkorderIds = userData.user?.bibliotekDkOrders?.map(
+        (order) => order.orderId
+      );
+      setBibliotekDkOrderIds(bibDkorderIds);
+    }
+  }, [userData]);
+
+  useEffect(() => {
+    //when a page is updated we fetch the next
+    const nextPageIndex = currentPageIndex + 1;
+    const toPages =
+      nextPageIndex == totalPages
+        ? bibliotekDkOrderIds.length - 1
+        : nextPageIndex;
+    const currentPageOrderIds = bibliotekDkOrderIds?.slice(
+      currentPageIndex * itemsPerPage,
+      toPages * itemsPerPage
+    );
+    setCurrentPageIds(currentPageOrderIds);
+  }, [currentPageIndex, bibliotekDkOrderIds, orderDataIsLoading]);
+
+  useEffect(() => {
+    if (fetchedOrders && !isLoading) {
+      let updatedOrderDataPages = orderDataPages;
+      updatedOrderDataPages[currentPageIndex] = fetchedOrders;
+      setOrderDataPages(updatedOrderDataPages);
+    }
+  }, [orderData, fetchedOrders, currentPageIds, currentPageIndex]);
+
+  //there is no pagination in mobile view. We show all orders
+  const currentPage = isMobile
+    ? orderDataPages.flat(1)
+    : orderDataPages[currentPageIndex];
+
+  return (
+    <Layout title={Translate({ context: "profile", label: "orderHistory" })}>
+      <Text className={styles.orderHistoryInfo}>
+        {Translate({ context: "profile", label: "orderHistoryInfo" })}
+      </Text>
+
+      <Link
+        onClick={() => {
+          modal.push("orderHistoryDataConsent");
+        }}
+        border={{
+          top: false,
+          bottom: {
+            keepVisible: true,
+          },
+        }}
+      >
+        <Text type="text3" tag="span">
+          {Translate({ context: "profile", label: "canWeSaveYourOrders" })}
+        </Text>
+      </Link>
+
+      <div className={styles.headerRow}>
```

Comment:
kræver en table ikke faktisk en del for at være accessible? det først jeg tænker på er de rette html-elementer som th og td. men der er nok en react måde at gøre det på

https://www.w3.org/WAI/tutorials/tables/